protocol/hash.go

```diff
@@ -39,6 +39,7 @@ const (
 	OneTimeSigKey2    HashID = "OT2"
 	PaysetFlat        HashID = "PF"
 	Payload           HashID = "PL"
+	Program           HashID = "Program"
```

Comment:
I haven't seen where you've been using this constant, but I might have missed it.

Reply:
Also, we use the `MX` for all arbitrary bytes. You may want to consider using it. If not, I'd consider changing the constant to follow the convention of 2-3 capital letters.

Reply:
@rotemh I think it should definitely have a new `HashID` since a program is semantically different from a human-readable message.

Reply:
I'm good with either.

Reply:
adam was suggesting that we should start to move to longer prefixes that might be more secure or give us more namespace or something, and it hardly matters how long they are since they're never stored anywhere